applications/GeoMechanicsApplication/tests/cpp_tests/test_thermal_dispersion_law.cpp

```diff
@@ -0,0 +1,115 @@
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//                   Gennady Markelov
+//
+
+#include "custom_constitutive/thermal_dispersion_law.hpp"
+#include "geo_mechanics_application.h"
+#include "includes/ublas_interface.h"
+#include "testing/testing.h"
+
+namespace Kratos::Testing {
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateThermalDispersionMatrix2D, KratosGeoMechanicsFastSuite)
+{
+    Model current_model;
+    auto& r_model_part = current_model.CreateModelPart("ModelPart", 1);
+
+    auto cond_prop = r_model_part.CreateNewProperties(0);
+    cond_prop->SetValue(POROSITY, 0.5);
+    cond_prop->SetValue(SATURATED_SATURATION, 0.75);
+    cond_prop->SetValue(RETENTION_LAW, "SaturatedLaw");
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_WATER, 1000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XX, 1500.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XY, 2000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_YY, 1200.0);
+
+    const SizeType dimension = 2;
+    GeoThermalDispersionLaw geo_thermal_dispersion2d_law(dimension);
+    ProcessInfo info;
+    Geometry<Node> geometry;
+
+    const Matrix thermal_dispersion_matrix =
+        geo_thermal_dispersion2d_law.CalculateThermalDispersionMatrix(
+            *cond_prop, info, geometry);
+
+    Matrix expected_solution = ZeroMatrix(2, 2);
+    expected_solution(0, 0) = 1125.0;
+    expected_solution(0, 1) = 1000.0;
+    expected_solution(1, 0) = expected_solution(0, 1);
+    expected_solution(1, 1) = 975.0;
+
+    constexpr double tolerance{1.0e-6};
+
+    for (unsigned int i = 0; i < thermal_dispersion_matrix.size1(); ++i) {
+        for (unsigned int j = 0; j < thermal_dispersion_matrix.size2(); ++j) {
+            KRATOS_EXPECT_NEAR(thermal_dispersion_matrix(i, j),
+                               expected_solution(i, j), tolerance);
+        }
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateThermalDispersionMatrix3D, KratosGeoMechanicsFastSuite)
+{
+    Model current_model;
+    auto& r_model_part = current_model.CreateModelPart("ModelPart", 1);
+
+    auto cond_prop = r_model_part.CreateNewProperties(0);
+    cond_prop->SetValue(POROSITY, 0.5);
+    cond_prop->SetValue(SATURATED_SATURATION, 0.75);
+    cond_prop->SetValue(RETENTION_LAW, "SaturatedLaw");
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_WATER, 800.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XX, 1000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XY, 2000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_YY, 1000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XZ, 1000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_YZ, 2000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_ZZ, 500.0);
+
+    const SizeType dimension = 3;
+    GeoThermalDispersionLaw geo_thermal_dispersion_2d_law(dimension);
```

Comment:
Since this is a test case for a 3D model, I would suggest to rename this variable:
```suggestion
    GeoThermalDispersionLaw geo_thermal_dispersion_3d_law(dimension);
```

Reply:
corrected

Reply:
the names are changed